swap/swap.go

```diff
@@ -78,9 +80,34 @@ func NewParams() *Params {
 	}
 }
 
+// NewLogger returns a new logger
+func NewLogger(h interface{}) log.Logger {
+	logLevel := 3
+	swapLogger := log.New("swaplog", "*")
+	hd, ok := h.(log.Handler)
+
+	if !ok {
+		//default handler just in case the handler is not set
+		hd = log.Root().GetHandler()
+		swapLogger.SetHandler(hd)
+		return swapLogger
+	}
+
+	//Filters messages with the correct logLevel for swap
+	hd = log.LvlFilterHandler(log.Lvl(logLevel), hd)
+
+	//Dispatches the logs to the default swarm log and also the filtered swap file logger.
+	swapLogger.SetHandler(log.MultiHandler(log.Root().GetHandler(), hd))
+
+	return swapLogger
+}
+
 // New - swap constructor
 func New(stateStore state.Store, prvkey *ecdsa.PrivateKey, backend contract.Backend) *Swap {
+	hd, _ := debug.SwapRotatingFileHandler()
```

Comment:
and finally, we can move `SwapRotatingFileHandler()` into this package and unexport it, and this would then just call `RotatingFileHandler()` - probably from the `debug` package, but we should think of where that goes finally too

Reply:
I left it separated, as maybe we want to change the split limit or format.

Reply:
fixed